code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -1736,3 +1736,44 @@
 	color = "#ED2939"
 	taste_description = "upside down"
 	can_synth = FALSE
+
+/// Improvised reagent that induces vomiting. Created by dipping a dead mouse in welder fluid.
+/datum/reagent/yuck
+	name = "Organic Slurry"
+	description = "A mixture of various colors of fluid. Induces vomiting."
+	glass_name = "glass of ...yuck!"
+	glass_desc = "It smells like a carcass, and doesn't look much better."
+	color = "#545000"
+	taste_description = "insides"
+	taste_mult = 4
+	can_synth = FALSE
+	metabolization_rate = 0.4 * REAGENTS_METABOLISM
+	var/yuck_cycle = 0 //! The `current_cycle` when puking starts.
+
+/datum/reagent/yuck/on_mob_add(mob/living/L)
+	if(HAS_TRAIT(src, TRAIT_NOHUNGER)) //they can't puke
+		holder.del_reagent(type)
+
+#define PUKE_CYCLES 3 		// every X cycle is a puke
+#define PUKES_TO_STUN 3 	// hit this amount of pukes in a row to start stunning
```

Comment:
put these at top and undefs on bottom of file. If you need to you can add yuck to clarify

Reply:
unless there's somewhere else that has it right above the block, idk how to enforce this so ill just use how other files do it.

Reply:
https://github.com/tgstation/tgstation/blob/21f7260651a62e2496418f42f32db68a92d7dabf/code/game/machinery/dance_machine.dm#L223-L301
https://github.com/tgstation/tgstation/blob/21f7260651a62e2496418f42f32db68a92d7dabf/code/game/objects/items/circuitboards/machine_circuitboards.dm#L149-L184
https://github.com/tgstation/tgstation/blob/21f7260651a62e2496418f42f32db68a92d7dabf/code/game/objects/items/robot/robot_items.dm#L339-L489
https://github.com/tgstation/tgstation/blob/21f7260651a62e2496418f42f32db68a92d7dabf/code/modules/atmospherics/machinery/portable/canister.dm#L220-L274

It just doesn't make sense to define for the entire file if it's only used in a specific part. Enforcing this is simple as: if someone uses it wrong the code won't compile and it can't be merged based off of that. If you want to know when they should undefine, then they should at the very least undefine at the end of the file but otherwise when they are done using it.

Reply:
seems sane

Reply:
Adding YUCK to the beginning for clarity is an excellent idea tho, I shoukd do that before merge. Can't tonight tho will tomorrow